lib/activerecord_spanner_adapter/connection.rb

```diff
@@ -186,33 +186,40 @@ def run_batch
 
     # DQL, DML Statements
 
-    def execute_query sql, params: nil, types: nil, transaction_required: nil
+    def execute_query sql, params: nil, types: nil
       if params
         converted_params, types = \
           Google::Cloud::Spanner::Convert.to_input_params_and_types(
             params, types
           )
       end
 
-      if transaction_required && !current_transaction&.active?
-        transaction = begin_transaction
-        self.current_transaction = transaction
-      elsif !current_transaction&.active?
+      # Clear the transaction from the previous statement.
+      unless current_transaction&.active?
         self.current_transaction = nil
       end
 
-      session.execute_query \
-        sql,
-        params: converted_params,
-        types: types,
-        transaction: transaction_selector,
-        seqno: (current_transaction&.next_sequence_number)
-    rescue Google::Cloud::AbortedError
-      # Mark the current transaction as aborted to prevent any unnecessary further requests on the transaction.
-      current_transaction&.mark_aborted
-      raise
-    ensure
-      commit_transaction if transaction
+      begin
+        session.execute_query \
+          sql,
+          params: converted_params,
+          types: types,
+          transaction: transaction_selector,
+          seqno: (current_transaction&.next_sequence_number)
+      rescue Google::Cloud::AbortedError => e
```

Comment:
'e' is not used for this block.

Reply:
Good catch, strange that `rubocop` did not complain about that.